### DISCUSSION TITLE
Why is gradlew setupdecomworkspace failling to open cd_settings

### DISCUSSION THREAD

**Discussion by GoodGolem** [maruohon/litematica#856] [Q&A]
![1](https://github.com/maruohon/litematica/assets/96292812/de725a87-c5a0-4b0e-bd37-fac9745714e9)

**Comment by sakura-ryoko** [comment]
You need to use the correct version of JDK
For 1.20.5+ Minecraft, you need Java 21.
For Minecraft 1.17 - 1.20.4 You need Java 17+

Also, coding is much easier using an IDE instead of the command prompt.
See https://fabricmc.net/wiki/tutorial:setup

**Reply by GoodGolem** [reply]
so java 22 wouldn't work right? also i am using vs code and it currently is importing a root project. is that normal?

**Comment by maruohon** [comment]
Let's take one step back. Are you making some changes to the code, or are you just looking for the 1.20.6 or 1.21 mod updates? There are already built mod jars for those versions in Sakura-Ryokos forked repos.

If you are working on the mod code, then which  MC version are you working in? Are you using the appropriate code branch as a starting point?

And finally `gradlew setupDecompWorkspace` is the old Gradle task from the 1.12.2 and older ForgeGradle-based mod tool chains (Forge or LiteLoader).

The Fabric tool chain's setup command is `gradlew genSources idea` or `gradlew genSources eclipse` (or if you want to use the VineFlower decompiler, like I do, then `gradlew genSourcesWithVineflower <idea | eclipse>`. If you are working on the recent Forge forks then I believe the command is just `gradlew idea` or `gradlew eclipse`? I can't remember for sure though. And I have no idea what it is for NeoForge.

And you need to be using the correct JDK version for the MC version. For 1.12.2 or older Forge or LiteLoader you need JDK 8. For Fabric you can use more recent JDK versions like JDK 16 up to MC 1.17.1 and MC 1.18+ I believe needs JDK 17+ and then 1.20.5+ needs JDK 21+. And the used Gradle plugin version (Loom for Fabric projects) also needs to be set appropriately to support the JDK version.

**Reply by GoodGolem** [reply]
oh there is alr litematica for 1.2.6? ok then i think the problem should be solved

**Reply by GoodGolem** [reply]
thank you very much. I am just gonna assume it is safe because idk any java code
